server/libs/websocket.js

```diff
@@ -0,0 +1,37 @@
+const WebSocket = require('ws');
+const db = require('./database');
+
+/**
+ * Function that handles setup and listening of websocket with the Angular app.
+ * Listens for a 'getData' event and sends the entire database back.
+ * 
+ * @param {server object from startup script} server 
+ */
+module.exports.socketSetupAndListen = function(server) {
+  const wss = new WebSocket.Server({server});
+
+  wss.on('connection', function(ws, req) {
+    // getData event handler
+    ws.on('getData', function(msg) {
```

Comment:
This is not really a different than a REST API and doesn't take advantage of websockets. The advantage websockets give us over a REST API is that the client doesn't need to ask, the server can just continuously give, but this doesn't really reflect that, the client shouldn't need to ask. The server should be broadcasting to all clients, all the time.